.github/workflows/imageupdate.yml

```diff
@@ -0,0 +1,30 @@
+name: Base image auto update
+on:
+  schedule:
+    - cron: "5 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up skopeo
+        run: sudo apt-get install -y skopeo
+      - name: Check ubi8-minimal change
+        run: |
+          ubiversion=$(skopeo inspect docker://registry.access.redhat.com/ubi8/ubi-minimal:latest |jq '.Labels.url' |sed 's/"//g' |awk -F/ '{print $NF}')
+      - name: Check go-toolset change
+        run: |
+          gotoolsetver=$(skopeo inspect docker://registry.access.redhat.com/ubi8/go-toolset:latest |jq '.Labels.url' |sed 's/"//g' |awk -F/ '{print $NF}')
+      - name: Do change if the version changed
+        run: |
+          git config user.name 'Update-a-Bot'
+          git config user.email 'webdev-spam-list <webdev-spam-list@redhat.com>'
+          sed -i "s/registry.access.redhat.com\/ubi8\/ubi-minimal:[0-9.-]*/registry.access.redhat.com\/ubi8\/ubi-minimal:$ubiversion/" Dockerfile
+          sed -i "s/registry.access.redhat.com\/ubi8\/go-toolset:[0-9.-]*/registry.access.redhat.com\/ubi8\/ubi-minimal:$gotoolsetver/" Dockerfile
+          git add -A
+          git commit -m "Update to ubi-minimal or go-toolset image" || echo "No changes to commit"
+      - name: Create pull request
+        run: gh pr create --title 'Regular docker image update' --body 'Automation \o/' --reviewer maknop
```

Comment:
I think we can heavily refactor this - potentially split it into 2 big blocks. 

One block is the changes made to the repo itself (that is - getting the newer version of the tags, then updating the tags accordingly)

Another block is to actually open a new PR against the repository with the suggested changes.

I'd say, first, let's open the PR using a GH action, there's multiple out there, I found this one for example: 

https://github.com/marketplace/actions/create-pull-request

then, put the logic that does all the finding the new version and replacing the data into the dockerfiles into a separate script. This brings the benefit of being able to run this locally (for testing and debugging purposes) and for better portability.